python/adbc_driver_manager/adbc_driver_manager/_lib.pyx

```diff
@@ -1006,26 +1088,31 @@ cdef class AdbcStatement(_AdbcHandle):
 
         Parameters
         ----------
-        data : int or ArrowArrayHandle
-        schema : int or ArrowSchemaHandle
+        data : PyCapsule or int or ArrowArrayHandle
+        schema : PyCapsule or int or ArrowSchemaHandle
         """
         cdef CAdbcError c_error = empty_error()
         cdef CArrowArray* c_array
         cdef CArrowSchema* c_schema
 
-        if isinstance(data, ArrowArrayHandle):
+        if PyCapsule_CheckExact(data):
```

Comment:
Would it make sense to look for and call the dunder method to convert to a PyCapsule?

Reply:
I first started doing that, but then decided to (for now) handle that on the DBAPI level, and here only deal with the actual pointers (whether wrapped in capsule, handle or as raw integer). 

It's easy to add, but then it makes the API a bit less consistent, because then `schema` would need to be made an optional argument, as that is embodied in the object with the dunder.

But given that the lower-level API exposes the dunders in the returned handles, it maybe also makes sense to let it ingest object with the dunders directly as well

Reply:
> Would it make sense to look for and call the dunder method to convert to a PyCapsule?

Updated to do that